src/mds/Server.cc

```diff
@@ -2994,7 +2997,7 @@ void Server::dispatch_peer_request(MDRequestRef& mdr)
     break;
 
   default: 
-    ceph_abort();
+    ceph_abort_msg("unknown op "+ to_string(op)+ " received");
```

Comment:
Hang on. I'm confused. Wan't this to be changed to blocklisting the client? I can see `evict_client` being called in `finish_reclaim_session` only.

Reply:
@vshankar The goal is to evict client when we see a client sending unknown msg. If its an MDS, its better to just abort it. Here if you see msgs are coming from MDS peer, its better to just abort it.

Reply:
And BTW, for now, evict is only happening in `handle_client_session`, I had put commit to evict client in `reclaim_session` but patrick and xiubo feel that we should just return `-EINVAL` to client. You can see that conversation here https://github.com/ceph/ceph/pull/47080#discussion_r932270561

Reply:
@vshankar What do you think about this https://github.com/ceph/ceph/pull/47080#issuecomment-1205148610?

Reply:
Thanks for explaining. I got a tiny bit confused with the github review UI not showing the function name.

Reply:
No problem!